tool/bump_version.dart

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'dart:io';
+
+void main() async {
+  await bumpVersionInAllPubspecFilesAndAddBlankEntryToChangelogs();
+}
+
+/// When executed, first prompts the user to enter old version, new version, and date.
+/// It then looks for instances of old version in pubspec files and updates them to new version.
+/// It also adds a blank entry to the top of CHANGELOG.md files.
+Future<void> bumpVersionInAllPubspecFilesAndAddBlankEntryToChangelogs() async {
+  Future<void> _addTextToTopOfFile(String filePath, String textToAdd) async {
+    final file = File(filePath);
+    var contents = await file.readAsString();
+    contents = '$textToAdd$contents';
+    await file.writeAsString(contents);
+  }
+
+  stdout.write('What is the old version number? \n');
+  String? oldVersion = stdin.readLineSync();
+
+  stdout.write('What is the new version number? \n');
+  String? newVersion = stdin.readLineSync();
+
+  stdout.write(
+      'What is the date for this release? Use international format yyyy-mm-dd e.g. 2020-07-29. \n');
+  String? date = stdin.readLineSync();
+  if (oldVersion != null && newVersion != null && date != null) {
+    // First, update in pubspec files.
+    final packagePubspecFilePattern = 'packages/**/pubspec.yaml';
+    final patternsForAllPubspecs = [
+      packagePubspecFilePattern,
+      'example/pubspec.yaml'
+    ];
+    // changes references to amplify libraries
+    patternsForAllPubspecs.forEach((pattern) async {
+      await Process.run('/bin/sh', [
+        '-c',
+        'sed -i "" "/amplify/s/$oldVersion/$newVersion/g" $pattern' // lines that have "amplify" and the old version, change old to new
+      ]);
+    });
+    // change the version: x at top of files
+    await Process.run('/bin/sh', [
+      '-c',
+      'sed -i "" "s/version: $oldVersion/version: $newVersion/g" $packagePubspecFilePattern' // instances of "version: <old>" to "version: <new>"
```

Comment:
Just wanna make sure there's no chance of collision with some other appearance of version? Afaik there isn't but figure you would know better at this time.

Reply:
I think there is some room for improvement here to ensure it occurs at start of line.

As written, could match something like "plugin_ends_with_version: 1.0.0" if it happens to have same version as `oldVersion`, although I guess it's somewhat unlikely.